tests/integration/pilot/mcs/discoverability/discoverability_test.go

```diff
@@ -141,8 +142,14 @@ func installMCSCRDs(t resource.Context) error {
 		if err != nil {
 			return err
 		}
-		if err := t.Config().ApplyYAML("", string(crd)); err != nil {
-			return err
+		if t.Settings().NoCleanup {
```

Comment:
Do we need an `ApplyYamlConditionalCleanup`?

Reply:
Yeah that might help a bit. We can always refactor later.